src/video-components/privateRoute/privateRoute.jsx

```diff
@@ -0,0 +1,13 @@
+import {Navigate,useLocation,Outlet} from "react-router-dom" 
+import { useAuth } from "../../context/authContext"
+
+export const PrivateRoute = () => {
+   const {token} = useAuth();
+   const location = useLocation();
+
+   return token ? (
+       <Outlet/>
```

Comment:
This is a new approach I got to know about Megha!! Good work on using library provided options for redirection tasks!!